src/common.speech/ServiceRecognizerBase.ts

```diff
@@ -214,6 +211,8 @@ export abstract class ServiceRecognizerBase implements IDisposable {
 
         this.privRecognizerConfig.SpeechServiceConfig.Context.audio = { source: deviceInfo };
 
+        // Start the connection to the service. The promise this will create is stored and will be used by configureConnection().
+        const conPromise: Promise<IConnection> = this.connectImpl();
```

Comment:
The connection was being opened before the audio was started intentionally to not serialize the two long running operations. Running them in series imparts a multi-hundred millisecond penalty.

Reply:
So should I restore the original order and wrap the audio start in a try/catch which closes (or undoes establishing steps for) the connection on error?

Reply:
Refactored as outlined in my previous comment.